### ISSUE TITLE
Support for AWS SAM templates

### ISSUE THREAD

**Issue by theseanything** [cloudreach/sceptre#324]
Would be great for native support of AWS SAM Templates using the Transform attribute.

This requires Sceptre:
 - Create a stack via change set
 - Package and deploy lambda code to S3

**Comment by theseanything** [comment]
Should solve issue #277 and #197

**Comment by thijsdev** [comment]
We would really like SAM / ChangeSets on create to be added.

**Comment by robertsigler** [comment]
> Package and deploy lambda code to S3

Maybe this is obvious, but I just wanted to point out that the aws cli already does this, so if we were willing to add `awscli` as a dependency, we might be able to lean on their implementation rather than having to rewrite it. https://github.com/aws/aws-cli/blob/develop/awscli/customizations/cloudformation/package.py

**Comment by zaro0508** [comment]
Hasn't this been implemented with the following command in V2?

```
create         Creates a stack or a change set
execute        Executes a Change Set
```

**Comment by zaro0508** [comment]
That's great @kkessels would you be able to propose a PR for the `CAPABILITY_AUTO_EXPAND` addition?

**Comment by kkessels** [comment]
I submitted the PR #586 for getting SAM templates working.

**Comment by jeshan** [comment]
@theseanything 
> Package and deploy lambda code to S3

Not available as at 2.1.0, right? (inline code works though)
Where are we tracking the relevant issue?

**Comment by jinwg** [comment]
how to use SAM template is really not clear, it seems we have to use the output generated by sam package command?
